src/components/PublicHeader.spec.tsx

```diff
@@ -13,7 +14,13 @@ describe('PublicHeader', () => {
     };
 
     it('contains a login button when not logged in', () => {
-        mount(<PublicHeader user={ null }/>);
+        mount(
+            <IntlProvider
+                locale="en"
+                messages={{ 'components.publicHeader.login_button': 'Login' }}>
+                <PublicHeader user={ null } />
+            </IntlProvider>,
+        );
```

Comment:
Using `IntlProvider` like this provides us with a very nice opportunity to control what strings are rendered. I wonder if it would be a good idea to test for the occurrence of those strings instead of using test ID attributes?